driver-sync/src/main/com/mongodb/client/internal/AggregateIterableImpl.java

```diff
@@ -167,15 +167,26 @@ public void toCollection() {
         }
     }
 
+    private BsonDocument getLastPipelineStage() {
+        if (lastPipelineStage == null) {
+            if (pipeline.isEmpty()) {
+                lastPipelineStage = new BsonDocument();
```

Comment:
I see it makes the callers easier to code by not returning null, but as a reader of the code it throws me off.   Would you consider returning null here and having callers check for null or empty pipeline explicitly?